### PR TITLE
Make sure package search exits with 1 or 0 in the SDK side

### PR DESCRIPTION
**Pull request by Nigusu-Allehu**
The NuGet side exits with
1 - if help output is required
2- if no help output is required but there was an error
0 - if success

This makes sure the SDK exits with 1 for the first two scenarios and 0 otherwise.
Related PR : https://github.com/NuGet/NuGet.Client/pull/5638
Related issue: https://github.com/NuGet/Home/issues/13161

### CODE REVIEW

**Comment by marcpopMSFT** [comment]
Should this target 8.0.3xx or are you planning to backport after?

**Review by nkolev92 — APPROVED** [review]

File: src/Cli/dotnet/commands/dotnet-package/search/PackageSearchCommandParser.cs [hunk 1]
```diff
@@ -97,6 +97,13 @@ private static CliCommand ConstructCommand()
                 {
                     parseResult.ShowHelp();
                 }
+
+                if (exitCode != 0)
+                {
+                    return 1;
+                }
+
+                return 0;
```
Comment (jeffkl) [inline src/Cli/dotnet/commands/dotnet-package/search/PackageSearchCommandParser.cs:106]: This _might_ look better?
```suggestion
                // Only return 1 or 0
                return exitCode == 0 ? 0 : 1;
```

**Comment by Nigusu-Allehu** [comment]
> Should this target 8.0.3xx or are you planning to backport after?

Planning to backport it

**Review by jeffkl — APPROVED** [review]

**Pull request merged**